ONSdigital/census-rm-ddl
# DDL Versioning

# Motivation and Context

We need to automate running database patches from our manual pipelines before deploying our apps. This PR adds a new `patch_database.py` script to allow us run patches to our databases using a provided `DDL_VERSION_TAG`. 

# What has changed

* Renamed existing patch filename to use underscores
* Added a script which will:
  * Check if database has already been patched with the provided tag (return if so)
  * Get all patch files for the current DDL version
  * Apply and record these patches in sequence after the MOST RECENT patch number
  * Update to the new version after success 


# How to test?



See output in our test pipeline in our dev area, or you could run this locally by creating the new schema and tweaking the config with necessary local bits


# Links



https://trello.com/c/fAB7azyN/403-ddl-versioning-13
https://github.com/ONSdigital/census-rm-kubernetes/pull/191
https://github.com/ONSdigital/census-rm-deploy/pull/123
